README.md

```diff
@@ -168,7 +180,9 @@ end
 ```
 
 ### Attributes:
-* ```directory```: The directory on Rackspace Cloud Files where the file can be found or should be uploaded to.
+* ```filename```: The local filepath to download the file to. This is the named_attribute. 
+* ```directory```: The container on Rackspace Cloud Files where the file can be found or should be uploaded to.
+* ```filepath```:  The path on Rackspace Cloud Files where the file can be found or uploaded to. This is the path under the container.  Eg. if the full path is my_container/my_subdirectory/my_file then filepath should be my_subdirectory/my_file.  If filepath is blank, the file is assumed to be in the root of the container, with the filename taken from the filename attribute.
```

Comment:
I think it's confusing already with this `filename` being local and the `directory` being remote(and actually a container), but the remote filepath being found from the local `named_attribute` basename :confused: 

> Or is that flexibility good to have when you don't want the file to be named the same locally and remotely?

I think it's a good feature. I think the syntax suggested by @martinb3 is good, to be honest I would even just have a `local` and `remote` path, with a sync/recursive feature (this PR feature and the sync/recursive is what prevented me to use this cookbook in the past).

For now, this PR seems like a great feature to be added, without breaking backward compatibility, maybe we should rename `filepath` so it's less confusing ? Maybe `container_filepath` ? It's annoying `directory` is not called `container`.